apis/s3/v1beta1/lifecycleConfiguration_types.go

```diff
@@ -0,0 +1,165 @@
+	AbortIncompleteMultipartUpload *AbortIncompleteMultipartUpload `json:"abortIncompleteMultipartUpload,omitempty"`
+
+	// Specifies the expiration for the lifecycle of the object in the form of date,
+	// days and, whether the object has a delete marker.
+	Expiration *LifecycleExpiration `json:"expiration,omitempty"`
+
+	// The Filter is used to identify objects that a Lifecycle Rule applies to.
+	// A Filter must have exactly one of Prefix, Tag, or And specified.
+	Filter *LifecycleRuleFilter `json:"filter,omitempty"`
+
+	// Unique identifier for the rule. The value cannot be longer than 255 characters.
+	ID *string `json:"ID,omitempty"`
+
+	// Specifies when noncurrent object versions expire. Upon expiration, Amazon
+	// S3 permanently deletes the noncurrent object versions. You set this lifecycle
+	// configuration action on a bucket that has versioning enabled (or suspended)
+	// to request that Amazon S3 delete noncurrent object versions at a specific
+	// period in the object's lifetime.
+	NoncurrentVersionExpiration *NoncurrentVersionExpiration `json:"noncurrentVersionExpiration,omitempty"`
+
+	// Specifies the transition rule for the lifecycle rule that describes when
+	// noncurrent objects transition to a specific storage class. If your bucket
+	// is versioning-enabled (or versioning is suspended), you can set this action
+	// to request that Amazon S3 transition noncurrent object versions to a specific
+	// storage class at a set period in the object's lifetime.
+	NoncurrentVersionTransitions []NoncurrentVersionTransition `json:"noncurrentVersionTransitions"`
+
+	// If 'Enabled', the rule is currently being applied. If 'Disabled', the rule
+	// is not currently being applied.
+	//
+	// Status is a required field, valid values are Enabled or Disabled
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	Status string `json:"status"`
+
+	// Specifies when an Amazon S3 object transitions to a specified storage class.
+	Transitions []Transition `json:"transitions"`
+}
+
+// AbortIncompleteMultipartUpload specifies the days since the initiation of an incomplete multipart upload
+// that Amazon S3 will wait before permanently removing all parts of the upload.
+// For more information, see Aborting Incomplete Multipart Uploads Using a Bucket
+// Lifecycle Policy (https://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html#mpu-abort-incomplete-mpu-lifecycle-config)
+// in the Amazon Simple Storage Service Developer Guide.
+type AbortIncompleteMultipartUpload struct {
+	// Specifies the number of days after which Amazon S3 aborts an incomplete multipart
+	// upload.
+	DaysAfterInitiation *int64 `json:"daysAfterInitiation,omitempty"`
+}
+
+// LifecycleExpiration contains for the expiration for the lifecycle of the object.
+type LifecycleExpiration struct {
+	// Indicates at what date the object is to be moved or deleted. Should be in
+	// GMT ISO 8601 Format.
+	Date *string `json:"date,omitempty"`
```

Comment:
```suggestion
	// Indicates at what date the object is to be moved or deleted.
	Date *metav1.Time `json:"date,omitempty"`
```

CRDs have support for time fields. This can be converted to Go `time.Time` easily.

Reply:
Oh thanks for the heads up, I had no idea this existed :)